project/merge_test.go

```diff
@@ -0,0 +1,148 @@
+package project
+
+import "testing"
+
+type NullLookup struct {
+}
+
+func (n *NullLookup) Lookup(file, relativeTo string) ([]byte, string, error) {
+	return nil, "", nil
+}
+
+func TestExtendsInheritImage(t *testing.T) {
+	p := NewProject(&Context{
+		ConfigLookup: &NullLookup{},
+	})
+
+	config, err := Merge(p, []byte(`
+parent:
+  image: foo
+child:
+  extends:
+    service: parent
+`))
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	parent := config["parent"]
+	child := config["child"]
+
+	if parent.Image != "foo" {
+		t.Fatal("Invalid image", parent.Image)
+	}
+
+	if child.Build != "" {
+		t.Fatal("Invalid build", child.Build)
+	}
+
+	if child.Image != "foo" {
+		t.Fatal("Invalid image", child.Image)
+	}
```

Comment:
I'd really, really, really like to use go-check.  I'm not a fan of testify and would like to move away from it.  I did start with testify for this code.  Are we cool with go-check?  Engine uses go-check